File: solutions/problem_2/solution.pl
use Time::HiRes qw( time );
sub solution {
  my ($sum, $a, $b) = (0, 1, 1);
  my $c = $a + $b;

  while($c < 4_000_000){
    $sum += $c;
    $a = $b + $c;
    $b = $a + $c;
    $c = $a + $b;
  }

  $sum;
}

my $start = time();
my $output = solution();
my $end = time();

print "=> Result: ", $output, "\n";
printf "=> Time: %fs\n", ($end - $start);

File: solutions/problem_2/solution.cpp
#include <iostream>
#include <iomanip>
#include <ctime>

int solution(){
    int s, a, b, c; 
    
    s = 0;
    a = 1;
    b = 1;
    c = a + b;

    while(c < 4000000){
        s += c;
        a = c + b;
        b = c + a;
        c = a + b;
    }

    return s;
}

int main(){
    std::clock_t start = std::clock();
    int output = solution();
    std::clock_t end = std::clock();

    std::cout << "=> Result: " << output << "\n";
    std::cout << std::fixed << std::setprecision(6) << "=> Time: "
              << (1000.0 * (end - start) / CLOCKS_PER_SEC) / 1000.0 << "s\n";

    return 0;
}